eternalcore-api/src/main/java/com/eternalcode/core/feature/ignore/event/IgnoreAllEvent.java

```diff
@@ -0,0 +1,57 @@
+package com.eternalcode.core.feature.ignore.event;
+
+import org.bukkit.entity.Player;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+import org.jetbrains.annotations.NotNull;
+
+public class IgnoreAllEvent extends Event implements Cancellable {
+
+    private static final HandlerList HANDLER_LIST = new HandlerList();
+
+    private final Player by;
+    private boolean cancelled;
+
+    public IgnoreAllEvent(@NotNull Player by) {
+        super(true);
+        this.by = by;
+    }
+
+    /**
+     * @return the player executing the ignore all action.
+     */
+
+    public Player getBy() {
+        return by;
+    }
+
+
+    /**
+     * @return Whether the event is cancelled
+     */
+
+    @Override
+    public boolean isCancelled() {
```

Comment:
Comment for the "getBy" method is good, but simple methods as isCancelled or setCancelled doesn't require a java docs comment.